app/javascript/components/firmware-registry-add-edit.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Grid, Row } from 'patternfly-react';
+import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-renderer';
+import MiqFormRenderer from '../forms/data-driven-form';
+
+const FirmwareRegistryAddEdit = (props) => {
+  const TYPES = [
+    ['FirmwareRegistry::RestApiDepot', __('Rest API Depot')],
+  ];
+
+  const [initialValues, setInitialValues] = useState({ type: TYPES[0][0] });
+
+  useEffect(() => {
+    props.dispatch({
+      type: 'FormButtons.init',
+      payload: {
+        newRecord: true,
+        pristine: true,
+      },
+    });
+    props.dispatch({
+      type: 'FormButtons.customLabel',
+      payload: __('Add'),
+    });
+  }, []);
+
+  const submitValues = (values) => {
+    API.post('/api/firmware_registries', {
```

Comment:
If you rename `username` name to `userid` in the schema, you can just do `values` here